src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -3112,3 +3283,68 @@ void java_bytecode_convert_methodt::create_stack_tmp_var(
   block.copy_to_operands(code_assignt(tmp_var, stack_entry));
   stack_entry=tmp_var;
 }
+
+/// Utility method to add a new symbol to the symbol table.
+/// Note: assumes that a symbol with this name does not exist.
+/// /param name: name of the symbol to be generated
+/// /param type: type of the symbol to be generated
+/// /param value: initial value of the symbol to be generated
+/// /param is_thread_local: if true this symbol will be set as thread local
+/// /param is_static_lifetime: if true this symbol will be set as static
+/// /return returns new symbol.
+symbolt java_bytecode_convert_methodt::add_new_symbol(
+  symbol_table_baset &symbol_table,
+  const irep_idt &name,
+  const typet &type,
+  const exprt &value,
+  const bool is_thread_local,
```

Comment:
These aren't very obvious at the callsite -- perhaps use two-valued enums so we don't mix up `true, false` and `false, true`?